__tests__/GestureHandler.test.js

```diff
@@ -0,0 +1,145 @@
+import React from 'react';
+import { render } from '@testing-library/react-native';
+import { Text } from 'react-native';
+import {
+  GestureHandlerRootView,
+  TapGestureHandler,
+  PanGestureHandler,
+  LongPressGestureHandler,
+  FlingGestureHandler,
+  RotationGestureHandler,
+  PinchGestureHandler,
+} from 'react-native-gesture-handler';
+import {
+  fireGestureHandlerTap,
+  fireGestureHandlerPan,
+  fireGestureHandlerLongPress,
+  fireGestureHandlerRotation,
+  fireGestureHandlerFling,
+  fireGestureHandlerPinch,
+  ghTagEventMacro,
+} from 'react-native-gesture-handler/src/jestUtils';
+import { useAnimatedGestureHandler } from '../src';
+
+const mockEventFunctions = () => {
+  return {
+    begin: jest.fn(),
+    progress: jest.fn(),
+    end: jest.fn(),
+  };
+};
+
+const assertEventCalls = (eventFunctions, progressCount = 1) => {
+  expect(eventFunctions.begin).toHaveBeenCalledTimes(1);
+  expect(eventFunctions.progress).toHaveBeenCalledTimes(progressCount);
+  expect(eventFunctions.end).toHaveBeenCalledTimes(1);
+};
+
+const App = (props) => {
+  const eventHandler = useAnimatedGestureHandler({
+    onStart: () => props.eventFunctions.begin(),
+    onActive: () => props.eventFunctions.progress(),
+    onEnd: () => props.eventFunctions.end(),
+  });
+
+  return (
+    <GestureHandlerRootView>
+      <TapGestureHandler onHandlerStateChange={eventHandler}>
+        <Text {...ghTagEventMacro()}>TapGestureHandlerTest</Text>
+      </TapGestureHandler>
+
+      <PanGestureHandler onHandlerStateChange={eventHandler}>
+        <Text {...ghTagEventMacro()}>PanGestureHandlerTest</Text>
+      </PanGestureHandler>
+
+      <LongPressGestureHandler onHandlerStateChange={eventHandler}>
+        <Text {...ghTagEventMacro()}>LongPressGestureHandlerTest</Text>
+      </LongPressGestureHandler>
+
+      <RotationGestureHandler onHandlerStateChange={eventHandler}>
+        <Text {...ghTagEventMacro()}>RotationGestureHandlerTest</Text>
+      </RotationGestureHandler>
+
+      <FlingGestureHandler onHandlerStateChange={eventHandler}>
+        <Text {...ghTagEventMacro()}>FlingGestureHandlerTest</Text>
+      </FlingGestureHandler>
+
+      <PinchGestureHandler onHandlerStateChange={eventHandler}>
+        <Text {...ghTagEventMacro()}>PinchGestureHandlerTest</Text>
+      </PinchGestureHandler>
+    </GestureHandlerRootView>
+  );
+};
+
+test('test fireGestureHandlerTap', () => {
+  const eventFunctions = mockEventFunctions();
+  const { getByText } = render(<App eventFunctions={eventFunctions} />);
+  fireGestureHandlerTap(getByText('TapGestureHandlerTest'));
```

Comment:
I'd like to suggest renaming this function to `fireTapGestureHandler` like the component name `<TapGestureHandler>` (and other functions as well).

Reply:
Sounds better 💪 I will rename these functions